packages/functions/src/index.ts

```diff
@@ -0,0 +1,98 @@
+import * as crypto from 'crypto';
+
+import * as admin from 'firebase-admin';
+import * as functions from 'firebase-functions';
+import * as nodemailer from 'nodemailer';
+
+admin.initializeApp();
+
+const SENDER_EMAIL = process.env.GMAIL_SENDER;
+const GMAIL_PASSWORD = process.env.GMAIL_PASSWORD;
+
+if (!SENDER_EMAIL || !GMAIL_PASSWORD) {
+  throw new Error('Sender email and password must be set as environment variables.');
+}
+
+const transporter = nodemailer.createTransport({
+  host:   'smtp.gmail.com',
+  port:   465,
+  secure: true,
+  auth:   {
+    user: SENDER_EMAIL,
+    pass: GMAIL_PASSWORD,
+  },
+});
+
+export const sendOTP = functions.https.onCall(
+  async (data: { email: string }, _context) => {
+    const { email } = data;
+
+    const otp = crypto.randomInt(100000, 999999).toString();
+
+    const otpDoc = admin.firestore().collection('otps').doc(email);
+    await otpDoc.set({
```

Comment:
As an enhancement, could we add a check to see if there is an existing non-expired OTP for the same email before creating a new one? This would help prevent flooding the otps collection with multiple OTP requests and ensure that old OTPs are not left without cleanup.

Reply:
There might be some issue with the email receiver that caused it to not receive the previous otp causing the user to click resend. I don't think it's good practice to re use the same code in these cases